zrml/prediction-markets/src/tests.rs

```diff
@@ -1125,6 +1125,160 @@ fn market_resolve_does_not_hold_liquidity_withdraw() {
     })
 }
 
+#[test]
+fn approve_market_correctly_unreserves_advisory_bond() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(PredictionMarkets::create_categorical_market(
+            Origin::signed(ALICE),
+            BOB,
+            MarketPeriod::Block(0..100),
+            gen_metadata(2),
+            MarketCreation::Advised,
+            2,
+            MarketDisputeMechanism::SimpleDisputes,
+            ScoringRule::CPMM,
+        ));
+        let alice_balance_before = Balances::free_balance(&ALICE);
+        assert_eq!(Balances::reserved_balance(&ALICE), AdvisoryBond::get() + OracleBond::get());
+        assert_ok!(PredictionMarkets::approve_market(Origin::signed(SUDO), 0));
+        assert_eq!(Balances::reserved_balance(&ALICE), OracleBond::get());
+        assert_eq!(Balances::free_balance(&ALICE), alice_balance_before + AdvisoryBond::get());
+    });
+}
+
+#[test]
+fn on_resolution_correctly_reserves_and_unreserves_bonds_for_permissionless_market_on_oracle_report()
```

Comment:
In a production environment, this function's name would indeed be too long. But we need the test description in the name of the test so that, during testing, we can easily see which tests fail. If you don't mind, I would like to keep the longer names.